UnityGameBase/Core/Utils/Extensions.cs

```diff
@@ -89,7 +89,12 @@ public static T GetRandomAndRemove<T>(this List<T> target)
         
         public static void Shuffle<T>(this List<T> list)
         {
-            System.Random rng = new System.Random();  
+            // System.Random() initialisation is obviously based on a low
+            // resolution timestamp of the current time, which yields
+            // identical shuffles if called in close succession.
+            // Solution: Use milliseconds, which provide more variation.
+            System.Random rng = new System.Random(System.DateTime.Now.Millisecond);
```

Comment:
This is reducing the entropy of the seed to 1000 possible values which is seriously bad. The default implementation uses Environment.TickCount (which is the number of milliseconds which passed since the machine started) which provides a much larger value set. Since the resolution is quite low, either add System.DateTime.Now.Millisecond, or (even better probably) System.DateTime.Now.Ticks (casted down to int). Another option could be to make the RNG a static member of the class, that way you don't have to instantiate it every time and you get new values every time you call Shuffle.